torch/nn/modules/upsampling.py

```diff
@@ -64,30 +45,48 @@ class UpsamplingNearest2d(_UpsamplingBase):
         [torch.FloatTensor of size 1x1x4x4]
 
     """
+    def __init__(self, size=None, scale_factor=None):
```

Comment:
Ya, I didn't like that redundancy either, but that doesn't seem to be the convention-  see for instance the dropout module and dropout function, which both ensure the value of P is between 0 and 1.

I can look into this, but where would you suggest such code go?  Seems like it'd set a precedent for refactoring lots of code.